src/App.tsx

```diff
@@ -1,61 +1,156 @@
+import { useEffect, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUser(userId: number): User | undefined {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || undefined;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedTitle, setTitle] = useState('');
+  const [selectedUser, setUser] = useState('');
+  const [addNewTodo, setTodo] = useState(todos);
+  const [validationTitle, setValTitle] = useState(true);
+  const [validationUser, setValUser] = useState(true);
+
+  const clearForm = () => {
+    setTitle('');
+    setUser('');
+  };
+
+  useEffect(() => {
+    return () => {
+      return clearForm();
+    };
+  }, [addNewTodo]);
```

Comment:
no needs to use this useEffect. Just move clearForm function to invoke after adding todo